engine/src/main/java/org/apache/apex/engine/YarnAppLauncherImpl.java

```diff
@@ -124,13 +121,15 @@ private void setConfiguration(Configuration conf, String property, Object value)
     }
   }
 
-  public class YarnAppHandleImpl implements YarnAppLauncher.YarnAppHandle
+  public class YarnAppHandleImpl implements YarnAppLauncher.YarnAppHandle, AutoCloseable
   {
     final ApplicationId appId;
+    final YarnClient yarnClient;
```

Comment:
Yes private is probably better even though it's being accessed by the parent class